apm-agent-core/src/test/java/co/elastic/apm/agent/report/ApmServerClientTest.java

```diff
@@ -243,23 +246,61 @@ public void testSimulateConcurrentConnectionError() {
     }
 
     @Test
-    public void testGetServerUrlsVerifyThatServerUrlsWillBeReloaded() throws IOException {
-        URL tempUrl = new URL("http", "localhost", 9999, "");
-        config.save("server_urls", tempUrl.toString(), SpyConfiguration.CONFIG_SOURCE_NAME);
+    public void testServerUrlsOverridesDefaultServerUrl() {
+        List<URL> updatedServerUrls = apmServerClient.getServerUrls();
+        // since only server_urls is set, we expect it to override the default server_url setting
+        assertThat(updatedServerUrls).isEqualTo(urlList);
+    }
 
+    @Test
+    public void testServerUrlsIsReloadedOnChange() throws IOException {
+        config.save("server_urls", "http://localhost:9999,http://localhost:9998", SpyConfiguration.CONFIG_SOURCE_NAME);
         List<URL> updatedServerUrls = apmServerClient.getServerUrls();
+        assertThat(updatedServerUrls).isEqualTo(List.of(
+            UrlValueConverter.INSTANCE.convert("http://localhost:9999"),
+            UrlValueConverter.INSTANCE.convert("http://localhost:9998")
+        ));
+    }
+
+    @Test
+    public void testDefaultServerUrls() throws IOException {
+        config.save("server_urls", "", SpyConfiguration.CONFIG_SOURCE_NAME);
+        List<URL> updatedServerUrls = apmServerClient.getServerUrls();
+        URL tempUrl = new URL("http", "localhost", 8200, "");
+        // server_urls setting is removed, we expect the default URL to be used
+        assertThat(updatedServerUrls).isEqualTo(List.of(tempUrl));
+    }
 
+    @Test
+    public void testServerUrlSettingOverridesServerUrls() throws IOException {
+        URL tempUrl = new URL("http", "localhost", 9999, "");
+        config.save("server_url", tempUrl.toString(), SpyConfiguration.CONFIG_SOURCE_NAME);
+        List<URL> updatedServerUrls = apmServerClient.getServerUrls();
         assertThat(updatedServerUrls).isEqualTo(List.of(tempUrl));
     }
 
+    @Test
+    public void testDisableSend() {
+        // We have to go through that because the disable_send config is non-dynamic
```

Comment:
Is there anything that would prevent this configuration from being dynamic ? If yes, then having a follow-up PR would be nice to prevent making this one larger.

Reply:
Making the `disable_send` config dynamic would be very easy, and even make its testing easier than it is now 🙂 
The reasons I thought it doesn't make sense to make it dynamic:
- when set to `true`, we are not polling for central config, which will make its dynamic nature quite impaired and confusing - it can be changed from `false` to `true` through central config, but not vice versa...
- the use case for this config is very specific -  you would normally use it when you don't want to setup APM server, but do want agent to trace things, for whatever reason. In such corner cases, I can't see a dynamic use for it. We already have the `recording` config which should be used in other cases.